lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1843,10 +1835,41 @@ static void checkClassConstructorBody(ClassDecl *classDecl,
   }
 }
 
+bool TypeCheckFunctionBodyAtLocRequest::evaluate(Evaluator &evaluator,
```

Comment:
I was under the impression we were going to define `TypeCheckFunctionBodyRequest` in terms of `TypeCheckFunctionBodyAtLocRequest`?

Reply:
I thought:
`TypeCheckFunctionBodyRequest` type checks the entire function body.
`TypeCheckFunctionBodyAtLocRequest` only type checks single body element at the location.

Reply:
Ah, I'm still thinking in terms of the old framework.

LGTM.